### PR TITLE
refactor: app refinement #P23-374

### PR DESCRIPTION
**Pull request by bdeluga**
# Overview 📖 
This refactor was undertaken due to the end of patronage and preparations for a demo. Here's a list:
- Removed mocks for budgets and transactions
- Merged two `types` folders into one placing it in `/lib`
- Components depending on more than 2 components were placed into folder with all of their imports.
- Moved default avatar src into `/avatars/default.svg`
- Moved validate functions `fixCurrencyObject.ts`,`iconValidation.ts`,`useValidateBudgetModal.tsx` into `lib/validations`
- Moved `Navbar` and `SideNavigationBar` to components 
- Changed `settings` page from `[...catchAll]` to correspoiding sub folders with their corresponding components, @zuzannapaw `fr:""` will help you find missing translates for French 🐸 ❤️ 
- Now when not logged you will be manually redirected to `sign-in` page (change [accepted ](https://intive-apu8934.slack.com/archives/C04R6LM4JSV/p1685285364110079)by client)

## Possible upgrade
There are still a lot of components that doesn't use superFetch, I can also refactor them if needed. 🚀 

### Concerns
I don't know how this PR will affect others work, and here's the question for leaders @wlechowicz @monika-klokosinska-intive. Should this be merged first or last in this sprint. I can pull main to this branch and resolve any conflicts there will be.

### CODE REVIEW

File: apps/storybook/stories/NavList.stories.tsx [hunk 1]
```diff
@@ -28,6 +34,48 @@ const dummyNavItemContentsTextOnly: Array<NavItemContents> = [
   },
 ];
 
+const dummyNavItemContentsWithIcon: Array<NavItemContents> = [
+  {
+    ComponentToRender: (
+      <>
+        <IconWrapper>
+          <Icon icon="payments" color="#1E4C40" />
```
Comment (wyola) [inline apps/storybook/stories/NavList.stories.tsx:42]: please use theme

(I just refuse to accept this PR until my opened PR is merged xD)

**Review by wyola — CHANGES_REQUESTED** [review]

File: apps/storybook/stories/NavList.stories.tsx (hunk 1 above)
Comment (bdeluga) [reply apps/storybook/stories/NavList.stories.tsx:42]: Is it even as possible as inline color  in storybook  ?

Comment (wyola) [reply apps/storybook/stories/NavList.stories.tsx:42]: Idk, it's a tactical move

**Review by wyola — APPROVED** [review]

**Review by zuzannapaw — APPROVED** [review]
Remarkable work! 🚀 Thank you so much for indicating places in dictionary for me, I completed it!🐸

**Review by PawelPazderski — APPROVED** [review]
Good job 🚀 I admire your patience and meticulousness!

**Review by Alessiass — APPROVED** [review]
Well done, I sense many conflicts to be resolved before the demo 🚀

**Review by Noshi96 — APPROVED** [review]
Great job on adjusting the folder structure, now it looks professional and clean 😎

**Review by Vectari — APPROVED** [review]
Great job! 🚀

**Review by wlechowicz — APPROVED** [review]
Good job.
Have fun resolving conflicts! There's already plenty, so it would be best to resolve it all here, merge this ASAP and then there should be limited changes required in each of the open PRs.

**Review by AzulloCyll — APPROVED** [review]
Great work! 🚀

**Review by wlechowicz — APPROVED** [review]
Must have been heaps of fun.

**Review by Noshi96 — COMMENTED** [review]
Something is wrong when logging in, it doesn't redirect me to the home/main page. The requests seem to work correctly. And after going directly to "/" by changing the URL, I can see that I am logged in.

**Comment by bdeluga** [comment]
> Something is wrong when logging in, it doesn't redirect me to the home/main page. The requests seem to work correctly. And after going directly to "/" by changing the URL, I can see that I am logged in.

Actually it redirects me  to `/` after login properly.

**Pull request merged**